module/zfs/zpl_xattr.c

```diff
@@ -478,8 +478,11 @@ zpl_xattr_set_sa(struct inode *ip, const char *name, const void *value,
 	}
 
 	/* Update the SA for additions, modifications, and removals. */
-	if (!error)
+	if (!error) {
 		error = -zfs_sa_set_xattr(zp);
+		if (error && (value != NULL))
+			(void) nvlist_remove(nvl, name, DATA_TYPE_BYTE_ARRAY);
```

Comment:
Is there a reason not to make this a VERIFY? This would be the only instance of nvlist_remove() in the module that doesn't either VERIFY or immediately test the error returned.